### ISSUE TITLE
Add parachute support

### ISSUE THREAD

**Issue by TatriX** [tdrhq/slite#1]
Hi!
Please consider adding [parachute](https://shinmera.github.io/parachute/) support.
Thanks!

**Comment by tdrhq** [comment]
Sure, let me take a look at it today, I might reach out if I have parachute specific questions since I haven't used it before

**Comment by tdrhq** [comment]
@TatriX Done! Try it out and let me know how it goes. I've updated the README with Parachute specific information. The only feature that doesn't work would be jumping to test definition. That would need a patch to Parachute, and even then would only work on Lispworks.

**Comment by tdrhq** [comment]
Things to look out for: In translating the results model from parachute to fiveam, I might've made some assumptions about the hierarchy of tests. Watch out for slite showing all green but parachute saying a test is failing. In this case, let me know what the hierarchy of the tests look like.

**Comment by TatriX** [comment]
When I'm trying to load `slite.el` I'm getting an error:
```
add-hook: Symbol’s function definition is void: case
```
from this code I think:
```
(add-hook (case (slite--slime-impl)
            (:sly
             'sly-compilation-finished-hook)
            (:slime
             'slime-compilation-finished-hook))
          'slite--compilation-finished)
 ```

**Comment by TatriX** [comment]
I think this is due `cl` lib being deprecated since Emacs 27 and probably it is remove in Emacs 28.
Most likely you want to `(require 'cl-lib)` and use `(cl-case ...)` as well as add `cl-` prefix to other functions as needed.

**Comment by tdrhq** [comment]
@TatriX Fixed: https://github.com/tdrhq/slite/commit/7ed03625eced85457b69675872e8bcbfd1a3db34

**Comment by TatriX** [comment]
Nice!
Now I do
```
(asdf:load-system :slite/parachute)
;; C-c v
(parachute:test :foo)
```
and getting:

```
Waiting for test results...
[sly] `sly-eval-async' errored: (void-function every)
```

**Comment by tdrhq** [comment]
Oops, git pull and try one more time: https://github.com/tdrhq/slite/commit/9a3bb24b9f066afd77ad5b7012f6b338e18f053a

**Comment by TatriX** [comment]
`define-test` name can be a symbol or a string:
`(define-test "A more descriptive name, perhaps")`

It seems this insn't handled right now:
```
The value
  "registered user should be able create an invite"
is not of type
  SYMBOL
   [Condition of type TYPE-ERROR]

Restarts:
 0: [*ABORT] Return to SLY's top level.
 1: [ABORT] abort thread (#<THREAD "slynk-worker" RUNNING {10058211C3}>)

Backtrace:
 0: ((:METHOD SLITE::GET-TEST-CASE-DETAILS (T)) #<TEST SERVER-TESTS::registered user should be able create an invite>) [fast-method]
 1: ((:METHOD SLITE::PROCESS-RESULTS (T)) #<PLAIN 14, FAILED results>) [fast-method]
 2: (SB-INT:SIMPLE-EVAL-IN-LEXENV (SLITE::PROCESS-RESULTS (EVAL (READ-FROM-STRING "(parachute:test 'server-tests)"))) #<NULL-LEXENV>)
 3: (EVAL (SLITE::PROCESS-RESULTS (EVAL (READ-FROM-STRING "(parachute:test 'server-tests)"))))
 --more--

```

**Comment by tdrhq** [comment]
@TatriX ah, let me think through this one

**Comment by tdrhq** [comment]
(keeping the issue opened for now)

**Comment by tdrhq** [comment]
@TatriX Okay done, try this one more time: https://github.com/tdrhq/slite/commit/766704b2179ee8a258e411f96aa4e4ecb57caddb

**Comment by tdrhq** [comment]
^ the above change will make the test results load, but rerunning in debugger appears to have broken with that change, will fix that in a second

**Comment by tdrhq** [comment]
Re-run in debugger fixed too https://github.com/tdrhq/slite/commit/df92f133f6a00d81bb88e9f2ce6d2e64065ca95c

**Comment by TatriX** [comment]
Now I'm getting:
```
[sly] `sly-eval-async' errored: (no-catch --cl-block-nil-- Failed: "(TRUE (SIGNIN (CURRENT-USER)))")
```

**Comment by tdrhq** [comment]
@TatriX at which point do you get this error, just when running the tests? Or was that when rerunning in the debugger?

**Comment by tdrhq** [comment]
Also, what was the CL expression you passed to `M-x slite-run`?

**Comment by TatriX** [comment]
I guess it will be easier to debug on my side. I used `C-c V` with `(parachute:test :my-package)`

**Comment by tdrhq** [comment]
@TatriX yeah, your debugging would be helpful in this situation :) In particular, here's what I'm looking for, if you run `(parachute:test :my-package)` in a repl, does it signal an error or does it just show you the failures? Is there some global parachute default that you've set that might cause failures to become conditions? (for instance, fiveam has fiveam:*on-failure* set to :signal or something like that, couldn't find anything obvious for parachute except for the specific report class)

The second issue is, even if that condition did get signalled, it should've just opened up a SLY debugger. Do you have any configuration that might prevent SLY from opening a debugger? For instance, if I run `M-x slite-run (parachute:test 'foobar :report 'parachute:interactive)` with a failing test, it still opens a debugger for me.

**Comment by ghost** [comment]
Hi,

There's a bug in `slite` that doesn't recognize failed parachute test results due to unhandled signal.

For the following code:
```
(defun foo ()
  (error "some error"))

(define-test should-fail
  (is = 1 1)
  (foo))
```

After  running `slite-run` with `(parachute:test 'should-fail)`, `slite` will report that the test passed:
```
PASS  SHOULD-FAIL                    1/1        
```

While the REPL report prints this:
```
        ？ CONDITION-FAIL-PARACHUTE::SHOULD-FAIL
  0.000 ✔   (is = 1 1)
WARNING: Unhandled error when evaluating CONDITION-FAIL-PARACHUTE::SHOULD-FAIL:
  some error

  0.003 ✘ CONDITION-FAIL-PARACHUTE::SHOULD-FAIL

;; Summary:
Passed:     1
Failed:     0
Skipped:    0

;; Failures:
   0/   1 tests failed in CONDITION-FAIL-PARACHUTE::SHOULD-FAIL
The test failed to evaluate properly:
  some error
#<PLAIN 2, FAILED results>
```

Just leaving it here in case anyone stumbles upon it as well.
